### PR TITLE
Update Source model based on current design

### PR DESCRIPTION
**Pull request by axmb**

### CODE REVIEW

File: verification/curator-service/api/src/model/source.ts [hunk 1]
```diff
@@ -31,31 +31,46 @@ const fieldSchema = new mongoose.Schema({
...
 });
 
+interface Parser {
+    awsLambdaArn: string;
+}
+
+const parserSchema = new mongoose.Schema({
+    awsLambdaArn: {
+        type: String,
+        required: 'Enter a parser AWS Lambda ARN',
```
Comment (khmoran) [inline verification/curator-service/api/src/model/source.ts:49]: do you plan on validating the format? (ex. the prefix arn:aws:us-east2)

File: verification/curator-service/api/src/model/source.ts [hunk 2]
```diff
@@ -31,31 +31,46 @@ const fieldSchema = new mongoose.Schema({
...
+    },
+});
+
+interface Schedule {
+    awsRuleArn: string;
+}
+
+const scheduleSchema = new mongoose.Schema({
```
Comment (khmoran) [inline verification/curator-service/api/src/model/source.ts:57]: design also mentions awsScheduleExpression for the cron -- adding later?

File: verification/curator-service/api/src/model/source.ts [hunk 3]
```diff
@@ -31,31 +31,46 @@ const fieldSchema = new mongoose.Schema({
...
+    awsRuleArn: {
```
Comment (khmoran) [inline verification/curator-service/api/src/model/source.ts:58]: same here -- planning to validate format (e.g. arn:aws:cronwatch prefix)?

File: verification/curator-service/api/src/model/source.ts [hunk 4]
```diff
@@ -31,31 +31,46 @@ const fieldSchema = new mongoose.Schema({
     },
 });
 
-interface Parsing {
+interface RegexParsing {
     fields: Array<Field>;
```
Comment (khmoran) [inline verification/curator-service/api/src/model/source.ts:35]: can this be [Field]? (for consistency w/ elsewhere)

File: verification/curator-service/api/src/model/source.ts [hunk 5]
```diff
@@ -17,45 +17,60 @@ const originSchema = new mongoose.Schema({
...
-        // TODO: Should probably use a validator for "cron(..." or "rate(...".
-        scheduleExpression: String,
-        parsing: parsingSchema,
-    },
+    parser: parserSchema,
+    schedule: scheduleSchema,
+    regexParsing: regexParsingSchema,
 });
 
 const sourceSchema = new mongoose.Schema({
```
Comment (khmoran) [inline verification/curator-service/api/src/model/source.ts:76]: q that doesn't need to be addressed in this PR -- but should format ultimately be an enum?

File: verification/curator-service/api/test/model/data/source.full.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "source name",
+    "origin": {
+        "url": "foo.bar",
+        "license": "MIT"
+    },
+    "format": "PDF",
+    "automation": {
```
Comment (khmoran) [inline verification/curator-service/api/test/model/data/source.full.json:8]: missing regexParsing?

File: verification/curator-service/api/test/model/source.test.ts [hunk 7]
```diff
@@ -1,20 +1,21 @@
 import { Source } from '../../src/model/source';
+import fullSource from './data/source.full.json';
+import minimalSource from './data/source.minimal.json';
```
Comment (khmoran) [inline verification/curator-service/api/test/model/source.test.ts:4]: maybe a test for Field to demonstrate you can't specify a Field without a regex? same maybe for Parser and awsLambdaArn, and Schedule and awsRuleArn?

File: verification/curator-service/api/src/model/source.ts (hunk 1 above)
Comment (axmb) [reply verification/curator-service/api/src/model/source.ts:49]: Yep -- had planned to add validations (and tests therearound) in a subsequent PR.

File: verification/curator-service/api/src/model/source.ts (hunk 2 above)
Comment (axmb) [reply verification/curator-service/api/src/model/source.ts:57]: Yeah, was planning to. I'm waffling on this a little; on one hand, it's a property of the rule, so I don't like incurring the cost of synchronizing it in our collection. On the other hand, I think it'll make the read/update flow simpler. Either way, was punting in this PR.

File: verification/curator-service/api/src/model/source.ts (hunk 3 above)
Comment (axmb) [reply verification/curator-service/api/src/model/source.ts:58]: Yep, per above.

File: verification/curator-service/api/src/model/source.ts (hunk 4 above)
Comment (axmb) [reply verification/curator-service/api/src/model/source.ts:35]: Good catch, will change this.

File: verification/curator-service/api/src/model/source.ts (hunk 5 above)
Comment (axmb) [reply verification/curator-service/api/src/model/source.ts:76]: That was my current understanding -- part of the planned validation.

File: verification/curator-service/api/test/model/data/source.full.json (hunk 6 above)
Comment (axmb) [reply verification/curator-service/api/test/model/data/source.full.json:8]: The reason it isn't here is that regexParsing should be mutually exclusive with a dedicated parser Lambda. Sorry, this is underexplained in this change; another one for the validation work.

File: verification/curator-service/api/test/model/source.test.ts (hunk 7 above)
Comment (axmb) [reply verification/curator-service/api/test/model/source.test.ts:4]: Yeah, we should; I was keeping this at the Source level, for now. WDYT about moving Origin/Automation to separate files, and adding detailed tests for each?

Comment (khmoran) [reply verification/curator-service/api/test/model/source.test.ts:4]: sgtm!

**Comment by khmoran** [comment]
LGTM, Approval 😆

**Comment by axmb** [comment]
[Create barebones source config UI backed by sources API/MongoDB](https://trello.com/c/aiNC8L8z/139-create-barebones-source-config-ui-backed-by-sources-api-mongodb)

**Comment by axmb** [comment]
[Implement curator service CRUD API for automated source scraping runs](https://trello.com/c/LOOSItyi/80-implement-curator-service-crud-api-for-automated-source-scraping-runs)

**Pull request merged**